stellargraph/connector/neo4j/mapper.py

```diff
@@ -54,11 +54,25 @@ class Neo4JBatchedNodeGenerator(BatchedNodeGenerator):
 
     def __init__(self, G, batch_size, neo4j_graphdb, schema=None):
 
-        import py2neo
-
         super().__init__(G, batch_size, schema)
+
+        try:
+            import py2neo
+        except ModuleNotFoundError as e:
+            raise ModuleNotFoundError(
+                f"{e.msg}. StellarGraph can only connect to Neo4j using the 'py2neo' module; please install it",
+                name=e.name,
+                path=e.path,
+            ) from None
+
         # Create neo4J driver
-        self.neo4j_graphdb = neo4j_graphdb
+        if not isinstance(neo4j_graphdb, py2neo.Graph):
+            raise TypeError(
+                f"neo4j_graphdb: expected py2neo.Graph, found {type(neo4j_graphdb)}"
+            )
+
+        else:
+            self.neo4j_graphdb = neo4j_graphdb
```

Comment:
I think it's better for this assignment to be unconditional. A strong convention in Python and in this library is to only use an `else` when it is necessary. For this code, it isn't necessary because execution of this function stops after the `raise` inside the `if`. That is:

```python
if not isinstance(...):
    raise TypeError(...)

self.neo4j_graphdb = neo4j_graphdb
```

Making the assignment unconditional is potentially slightly nicer from a code analysis/tooling perspective too, because then it's "obvious" that this `neo4j_graphdb` property is always assigned and always exists.